src/main/java/com/couchbase/lite/Attachment.java

```diff
@@ -124,15 +125,36 @@ public String getContentType() {
     public InputStream getContent() throws CouchbaseLiteException {
         if (body != null) {
             return body;
-        }
-        else {
+        } else {
             Database db = revision.getDatabase();
-            Attachment attachment = db.getAttachmentForSequence(revision.getSequence(), this.name);
+            long sequence = getAttachmentSequence();
+            if (sequence == 0) {
+                throw new CouchbaseLiteException(Status.INTERNAL_SERVER_ERROR);
+            }
+            Attachment attachment = db.getAttachmentForSequence(sequence, this.name);
             body = attachment.getContent();
+            if (attachment.getGZipped()) {
```

Comment:
Router uses Databse.getAttachmentForSequence to retrieve the attachment which uses the constructor with the InputStream. This else clause is only reached when instantiated through a Revision.